go/common/cbor/codec.go

```diff
@@ -40,15 +37,16 @@ type MessageReader struct {
 }
 
 // Read deserializes a single CBOR-encoded Message from the underlying reader.
-func (c *MessageReader) Read(msg interface{}) error {
+func (c *MessageReader) Read(msg interface{}, caller string) error {
```

Comment:
Uh I don't like this, wouldn't the caller always be the same for the lifetime of the MessageReader? If anything it should be a field in the MessageReader.